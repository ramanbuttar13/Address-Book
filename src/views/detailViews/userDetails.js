import React from 'react';
import { Link } from 'react-router-dom'
import {
  Box,
  Flex,
  Text,
  Breadcrumb,
  BreadcrumbItem,
  BreadcrumbLink,
  Button
} from '@chakra-ui/react';
//Store
import {connect} from 'react-redux';

//UI components
import Header from "../../components/Header";
import Footer from "../../components/Footer";
import theme from '../../components/styles/theme'
import Avatar from '../../components/User/Avatar'



const mapStateToProps = (state) => {
	return {
		users: state.homepage.users,
	};
};


export const PeopleDetails = ({users, match, abs = 'undefined'}) => {
  const userId = match.params.id
  // -1 because redux store starts from 0 whereas the random user is 1 indexed
  const user = users[userId-1];

  return(
    <>
    <Header page="people"/>
    {
      user ? <>
      <Box m="16px">
      <Breadcrumb spacing="8px">
        <BreadcrumbItem>
          <Link to="/"><BreadcrumbLink href="#">Homepage</BreadcrumbLink></Link>
        </BreadcrumbItem>
        <BreadcrumbItem isCurrentPage>
          <BreadcrumbLink href="#">{user.name.first}</BreadcrumbLink>
        </BreadcrumbItem>
      </Breadcrumb>
    </Box>
    <Box margin={["20px auto", "100px auto"]} width={["fit-content","fit-content", "600px"]} position="relative" p="8px" textAlign="center" border={"1px solid "+theme.colors.brand.primary} borderRadius={theme.radii.lg}>
      <Flex textAlign="left" flexDir={["column", "column", "row"]}>
        <Box my="auto">
          <Avatar radius="10px" src={user.picture.large} avatarSize="300px"/>
        </Box>
        <Flex ml="16px" flexDirection="column">
          <Box py="4px">
            <Text fontWeight="bold"> Firstname: &nbsp;</Text>
            <Text> {user.name.first}</Text>
          </Box>
          <Box>
            <Text fontWeight="bold"> Lastname: &nbsp;</Text>
            <Text> {user.name.last}</Text>
          </Box>
          <Box>
            <Text fontWeight="bold"> Phone number: &nbsp;</Text>
            <Text> {user.phone}</Text>
          </Box>
          <Box>
            <Text fontWeight="bold"> Age: &nbsp;</Text>
            <Text> {user.dob.age}</Text>
          </Box>
          <Box>
            <Text fontWeight="bold"> Email: &nbsp;</Text>
            <Text> {user.email}</Text>
          </Box>
          <Box>
            <Text fontWeight="bold"> Gender: &nbsp;</Text>
            <Text> {user.gender}</Text>
          </Box>
        </Flex>
        </Flex>
      </Box>
      </>: 
      <Box m="16px">
        <Text my="8px">Oops! something went wrong.</Text>
        <Link to="/">
        <Button>Go Back to Homepage</Button>
        </Link>
      </Box>}
    <Footer/>
    </>
  )
}

export default connect(mapStateToProps)(PeopleDetails);