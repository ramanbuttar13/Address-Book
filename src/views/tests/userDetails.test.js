// Link.react.test.js
import React from 'react';
import renderer from 'react-test-renderer';
import { PeopleDetails } from '../detailViews/userDetails.js';
import { createMemoryHistory } from 'history'
import { Router } from 'react-router-dom';
import { users, match } from './constants';

test('Should run User Details snapshot test', () => {
	const history = createMemoryHistory()
  const component = renderer.create(
    <Router history={history}>
		  <PeopleDetails users={users} match={match} />
    </Router>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
