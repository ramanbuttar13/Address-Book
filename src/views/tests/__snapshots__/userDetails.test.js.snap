// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
Array [
  <div
    className="sc-bdnxRM dQyMqc"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <p
        className="chakra-text css-u5d0r8"
      >
        Nuvalence
      </p>
    </a>
  </div>,
  <div
    className="css-1c5ij41"
  >
    <p
      className="chakra-text css-1xlzx9v"
    >
      Oops! something went wrong.
    </p>
    <a
      href="/"
      onClick={[Function]}
    >
      <button
        className="chakra-button css-5c2176"
        type="button"
      >
        Go Back to Homepage
      </button>
    </a>
  </div>,
  <div
    className="sc-gtsrHT fYTXOB"
  >
    Made by: Raman Buttar
  </div>,
]
`;

exports[`Should run User Details snapshot test 1`] = `
Array [
  <div
    className="sc-bdnxRM dQyMqc"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <p
        className="chakra-text css-u5d0r8"
      >
        Nuvalence
      </p>
    </a>
  </div>,
  <div
    className="css-1c5ij41"
  >
    <p
      className="chakra-text css-1xlzx9v"
    >
      Oops! something went wrong.
    </p>
    <a
      href="/"
      onClick={[Function]}
    >
      <button
        className="chakra-button css-5c2176"
        type="button"
      >
        Go Back to Homepage
      </button>
    </a>
  </div>,
  <div
    className="sc-gtsrHT fYTXOB"
  >
    Made by: Raman Buttar
  </div>,
]
`;
